### PR TITLE
Provide JSON output of player info on HTTP request for /player

### PR DESCRIPTION
**Pull request by Gigabyte5671**
Provides a way to retrieve basic info for a player requested with the URL query string ( /player?NAME ).  

Player info includes:
- Player name,
- Clan tag,
- Rank XP.

### CODE REVIEW

File: login_server/loginserver.py [hunk 1]
```diff
@@ -351,9 +352,42 @@ def handle_http_request_message(self, msg):
...
+                msg.peer.send_response(json.dumps({
+                    'player_found': False
+                }, sort_keys=True, indent=4))
         else:
             msg.peer.send_response(None)
 
+    def get_player_settings_data(self, player_name):
+        self.logger.info("Retrieved settings data for player: " + player_name)
+        try:
+            with open(glob.glob('data/players/' + player_name + '_settings.json')[0], "r") as f:
```
Comment (Griffon26) [inline login_server/loginserver.py:385]: Why would you use glob here when you already know the filename? Do you want people to pass wildcards?

Using input from a user to determine a file to read is very unsafe. What if I were to specify "../gamesettings/ootb/serverconfig.lua\0" in order to read the admin password. I know it requires more to really do anything harmful, but this is the first step.

The input data should not be trusted. So we could check if it's a valid username by checking if it is in self.accounts.

File: login_server/loginserver.py [hunk 2]
```diff
@@ -351,9 +352,42 @@ def handle_http_request_message(self, msg):
...
+                file_contents = json.load(f)
+            return file_contents
+        except IndexError:
+            return False
```
Comment (Griffon26) [inline login_server/loginserver.py:389]: use None instead of False

File: login_server/loginserver.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
 import logging
 import random
 import string
+import glob
```
Comment (Griffon26) [inline login_server/loginserver.py:27]: Remove this

**Review by Griffon26 — CHANGES_REQUESTED** [review]

**Comment by Gigabyte5671** [comment]
As per requested changes:
- Removed use of glob.
- Filter incoming player name request and check it against self.accounts before processing.
- Return None instead of False on the exception that the player settings file is not found.

See [076db4d](https://github.com/Griffon26/taserver/pull/114/commits/076db4da8a3666b4e427b3d4246793d33640d4d5)

File: login_server/loginserver.py [hunk 4]
```diff
@@ -351,9 +351,44 @@ def handle_http_request_message(self, msg):
...
             }, sort_keys=True, indent=4))
+        elif msg.env['PATH_INFO'] == '/player':
+            if "REMOTE_ADDR" in msg.env:
+                self.logger.info('Served player stats request via HTTP to peer "' + msg.env["REMOTE_ADDR"] + '"')
+            else:
+                self.logger.info('Served player stats request via HTTP to Unknown peer')
+            
+            if "QUERY_STRING" in msg.env:
+                filtered_player_name = ''.join(filter(str.isalnum, msg.env['QUERY_STRING']))
+                player_data = False
```
Comment (Griffon26) [inline login_server/loginserver.py:362]: None fits better

File: login_server/loginserver.py [hunk 5]
```diff
@@ -351,9 +351,44 @@ def handle_http_request_message(self, msg):
...
+                    msg.peer.send_response(json.dumps({
+                        'player_found': False
+                    }, sort_keys=True, indent=4))
+            else:
+                msg.peer.send_response(None)
         else:
             msg.peer.send_response(None)
 
+    def get_player_settings_data(self, player_name):
+        self.logger.info("Retrieved settings data for player: " + player_name)
```
Comment (Griffon26) [inline login_server/loginserver.py:384]: Do we care about this enough to log it?

Comment (Gigabyte5671) [reply login_server/loginserver.py]: Good point. "Served player stats request..." is already logged, this log is triggered by the same set of functions so is probably unnecessary. I've removed this.

**Comment by Gigabyte5671** [comment]
As per requested changes:
- Change default value of player_data from False to None.
- Remove unnecessary extra log.

See [38fd8d1](https://github.com/Griffon26/taserver/pull/114/commits/38fd8d188583318648686ddd606ad2bf89dca1ed) above.

**Review by Griffon26 — APPROVED** [review]

**Pull request merged**